pkg/workloads/cri.go

```diff
@@ -61,9 +62,10 @@ type criClient struct {
 	criRuntime.RuntimeServiceClient
 
 	endpointManager *endpointmanager.EndpointManager
+	allocator       *cache.IdentityAllocatorManager
 }
 
-func newCRIClient(ctx context.Context, epMgr *endpointmanager.EndpointManager) (*criClient, error) {
+func newCRIClient(ctx context.Context, epMgr *endpointmanager.EndpointManager, allocator *cache.IdentityAllocatorManager) (*criClient, error) {
```

Comment:
again, having the `IdentityAllocatorManager` part of the endpoint would allow to not having this file touched.